variables.tf

```diff
@@ -2,8 +2,72 @@
 # Input Variables
 ########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "watsonx_data_name" {
+  type        = string
+  description = "The name of the watsonx.data instance."
+  default     = null
+
+  validation {
+    condition     = var.existing_watsonx_data_instance_crn == null ? length(var.watsonx_data_name) > 0 : true
+    error_message = "You must specify a value for 'watsonx_data_name' if 'existing_watsonx_data_instance_crn' is null."
+  }
+}
+
+variable "resource_group_id" {
+  type        = string
+  description = "The resource group ID where the watsonx data instance is created."
+  default     = null
```

Comment:
Validation should be added here to ensure this value is not `null` when `existing_watsonx_data_instance_crn` is `null`